liquibase-core/src/main/java/liquibase/database/core/PostgresDatabase.java

```diff
@@ -330,9 +338,22 @@ protected SqlStatement getConnectionSchemaNameCallStatement() {
         return new RawCallStatement("select current_schema()");
     }
 
+    /**
+     * Generates PK following {@code PostgreSQL} conventions:
+     * <li>Postgres PK size is limited with 63 bytes.
+     * <li>Postgres PK is suffixed with '_pkey'.
+     *
+     * @param tableName Table name as the base name for the generated PK.
+     * @return PK name.
+     */
     @Override
     public String generatePrimaryKeyName(final String tableName) {
-        return tableName.toUpperCase(Locale.US) + "_PKEY";
+        final byte[] bytes = tableName.getBytes(StandardCharsets.UTF_8);
+        final int suffixBytes = PGSQL_PK_SUFFIX.length(); // All characters are ASCII symbols => 1 byte per symbol.
```

Comment:
If the assumption here is to use UTF-8 then that is fine but I would still probably define a private method parameterized by Charset that is driven by that so any future enhancement to make the code gracefully handle other charsets is done easily and with little opportunity for error.

Reply:
If not then I'd just declare a static final byte[] for the UTF-8 encoded "_pkey" suffix and use that here.

Reply:
Good suggestions.